File: DictoData/Model/Word.cs
using System.Collections.Generic;
using DictoData.Core;
using DictoInfrasctructure.Enums;

namespace DictoData.Model
{
    public class Word : CoreEntity
    {

        public Word()
        {
        }
        
        public string Text { get; set; }

        public LevelType Level { get; set; }

        public string Phonetic { get; set; }
        
        public SuperMemory SuperMemory { get; set; }

        public int UserId { get; set; }

        public User User { get; set; }

        public virtual List<Translate> Translates { get; set; }

        public virtual List<Example> Examples { get; set; }

        public int? DeckId { get; set; }

        public Deck Deck { get; set; }
        
    }
}

File: client-spa/src/app/models/word.ts
import {Translate} from './translate';
import {Deck} from './deck';
import {Example} from './example';



export enum LevelType {
  None = 0,
  First = 1,
  Second = 2,
  Third = 3,
  Compleate = 4
}


export class Word {
  public id: number;
  public created: string;
  public text: string;
  public level: LevelType;
  public phonetic: string;
  public userid: number;
  public translates: Translate[];
  public examples: Example[];
  public deckid: number;
  public deck: Deck;
  public sound: string;
}
